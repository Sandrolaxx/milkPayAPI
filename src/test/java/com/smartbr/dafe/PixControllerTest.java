package com.smartbr.dafe;

import io.quarkus.test.common.QuarkusTestResource;
import io.quarkus.test.junit.QuarkusTest;
import io.restassured.RestAssured;
import io.restassured.http.ContentType;
import io.restassured.http.Header;
import io.restassured.specification.RequestSpecification;

import org.approvaltests.Approvals;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import com.github.database.rider.cdi.api.DBRider;
import com.github.database.rider.core.api.configuration.DBUnit;
import com.github.database.rider.core.api.configuration.Orthography;
import com.github.database.rider.core.api.dataset.DataSet;
import com.smartbr.dafe.util.TokenUtils;

@DBRider
@DBUnit(caseInsensitiveStrategy = Orthography.LOWERCASE)
@QuarkusTest
@QuarkusTestResource(DafeTestLifecycleManager.class)
public class PixControllerTest {

    private String token;

    @BeforeEach
    public void genereteToken() throws Exception {
        token = TokenUtils.generateTokenString("/JWTDafePay.json", null);
    }

    private RequestSpecification given() {
        return RestAssured.given()
        .contentType(ContentType.JSON).header(new Header("Authorization", "Bearer " + token));
    }


    @Test
    @DataSet("cenario-testCase02.json")
    public void testConsultPix() {
        String result = given()
            .header("txId", 1)
            .header("provider", "celcoin")
            .when().get("/dafe-api/v1/pix/static")
            .then()
            .statusCode(200)
            .extract().asString();

        Approvals.verifyJson(result);
    }


}