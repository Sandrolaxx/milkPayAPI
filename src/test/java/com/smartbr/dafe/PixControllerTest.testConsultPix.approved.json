{
  "amount": 4.0,
  "endToEndId": "E08744817202108101254LZXLTQRKKAS",
  "liquidatedDigitalAccount": true,
  "paymentDate": "2021-06-30 16:23:05",
  "status": "PAID",
  "transactionType": "RECEIVEPIX",
  "txId": "1"
}